### PR TITLE
Chat: indigo join page

### PR DESCRIPTION
**Pull request by tacryt-socryp**
Does this look good?

### CODE REVIEW

File: pkg/interface/src/apps/chat/components/join.js [hunk 1]
```diff
@@ -2,145 +2,99 @@ import React, { Component } from 'react';
...
-        station.length < 2 ||
-        !urbitOb.isValidPatp(ship)
-      ) {
-        this.setState({
-          error: true
-        });
+  onSubmit(values) {
+    console.log(values);
+    this.setState({ awaiting: true }, () => {
+      console.log(values);
```
Comment (matildepark) [inline pkg/interface/src/apps/chat/components/join.js:43]: Remove console.logs please

File: pkg/interface/src/apps/chat/components/join.js [hunk 2]
```diff
@@ -91,56 +88,45 @@ export class JoinScreen extends Component {
...
-            className={joinClasses}
-          >Join Chat</button>
-          <Spinner awaiting={this.state.awaiting} classes="mt4" text="Joining chat..." />
-        </div>
-      </div>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={schema}
+        onSubmit={this.onSubmit.bind(this)}>
+        <Form>
```
Comment (g-a-v-i-n) [inline pkg/interface/src/apps/chat/components/join.js:64]: Looks good, and I can see why we need Form to have styled props

File: pkg/interface/src/apps/chat/components/join.js [hunk 3]
```diff
@@ -91,56 +88,45 @@ export class JoinScreen extends Component {
...
+          <Box width="100%" height="100%" p={3} overflowX="hidden">
+            <Box
+              width="100%"
+              pt={1} pb={5}
+              display={['', 'none', 'none', 'none']}
+              fontSize={0}>
+              <Link to="/~chat/">{'⟵ All Chats'}</Link>
+            </Box>
+            <Text mb={3} fontSize={0}>Join Existing Chat</Text>
+            <Box width="100%">
```
Comment (g-a-v-i-n) [inline pkg/interface/src/apps/chat/components/join.js]: This will get a lot easier in indigo-react 1.2 with <Inline />

**Comment by matildepark** [comment]
Conflict with the new organisation -_-

**Pull request merged**